### PR TITLE
Idempotent producer example 

### PR DESCRIPTION
**Pull request by edenhill**
And sneaked in a release checksum script.

### CODE REVIEW

File: src/rdkafka.h [hunk 1]
```diff
@@ -673,6 +673,27 @@ rd_kafka_resp_err_t rd_kafka_fatal_error (rd_kafka_t *rk,
                                           char *errstr, size_t errstr_size);
 
 
+/**
+ * @brief Trigger a fatal error for testing purposes.
```
Comment (mhowlett) [inline src/rdkafka.h]: ideally I think this functionality should be tested via integration tests using a mock broker, or intercepting and modifying calls to a real broker - I don't like the idea of having methods in a public API for testing purposes. I admit though that is difficult to set up and that the testing in this PR brings value. In the event we start doing integration testing as described above though, I think we should get rid of this method. so could maybe mark as such, but just my opinion.

Comment (edenhill) [reply src/rdkafka.h]: As stated elsewhere, it is practically impossible to trigger idempotent producer fatal errors without a full mock broker.

I think this API serves an important purpose for our users to allow them to easily test their (error handling) code, we wouldn't want to require them setting up a eos-breaking mock broker just to verify their code logic.

File: examples/rdkafka_idempotent_producer.c [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+ * This callback is triggered by rd_kafka_poll() or rd_kafka_flush()
+ * for client instance-level errors, such as broker connection failures,
+ * authentication issues, etc.
+ *
+ * These errors should generally be considered informational as
+ * the underlying client will automatically try to recover from
+ * any errors encountered, the application does not need to take
+ * action on them.
+ *
+ * But with idempotence truly fatal errors can be raised when
```
Comment (mhowlett) [inline examples/rdkafka_idempotent_producer.c]: comma after idempotence [you can rely on my to pick up on the important stuff]

File: examples/rdkafka_idempotent_producer.c [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+ * the idempotence guarantees can't be satisfied, these errors
```
Comment (mhowlett) [inline examples/rdkafka_idempotent_producer.c]: new sentence after satisfied, not comma.

File: examples/rdkafka_idempotent_producer.c [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+                                 * The internal queue represents both
+                                 * messages to be sent and messages that have
+                                 * been sent or failed, awaiting their
+                                 * delivery report callback to be called.
+                                 *
+                                 * The internal queue is limited by the
+                                 * configuration property
+                                 * queue.buffering.max.messages */
+                                rd_kafka_poll(rk, 1000/*block for max 1000ms*/);
+                                goto retry;
```
Comment (mhowlett) [inline examples/rdkafka_idempotent_producer.c]: Interesting that you prefer goto. seems justifiable - one level less nesting, less loc, not spaghetti here.

File: examples/rdkafka_idempotent_producer.c [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+                 * you register). */
+                rd_kafka_poll(rk, 0/*non-blocking*/);
+
+                msgcnt++;
+
+                /* Since fatal errors can't be triggered in practice,
+                 * use the test API to trigger a fabricated error after
+                 * some time. */
+                if (msgcnt == 13)
+                        rd_kafka_test_fatal_error(
```
Comment (mhowlett) [inline examples/rdkafka_idempotent_producer.c]: I don't think an example should contain a test method.

File: examples/rdkafka_idempotent_producer.c (hunk 4 above)
Comment (edenhill) [reply examples/rdkafka_idempotent_producer.c]: Yep, those are the reasons. :+1:

File: examples/rdkafka_idempotent_producer.c (hunk 5 above)
Comment (edenhill) [reply examples/rdkafka_idempotent_producer.c]: I believe it is important to show our users how to test their code.

File: src/rdkafka.h (hunk 1 above)
Comment (mhowlett) [reply src/rdkafka.h]: very good point that users would find this useful for testing. ok, sounds good to me.

File: examples/rdkafka_idempotent_producer.c (hunk 5 above)
Comment (edenhill) [reply examples/rdkafka_idempotent_producer.c]: It is also very unlikely that people will leave this as-is in the code as they move the example over to their real code.

**Pull request merged**